R/ttestindependentsamples.R

```diff
@@ -618,3 +619,109 @@ ttestIndependentMainTableRow <- function(variable, dataset, test, testStat, effS
   
   return(p)
 }
+
+.ttestIndependentDescriptivesRainCloudPlot <- function(jaspResults, dataset, options, ready) {
+  if(!options$descriptivesPlotsRainCloud)
+    return()
+  .ttestDescriptivesContainer(jaspResults, options)
+  container <- jaspResults[["ttestDescriptives"]]
+  container[["plotsRainCloud"]] <- createJaspContainer(gettext("Raincloud-like Plots"))
+  subcontainer <- container[["plotsRainCloud"]]
+  subcontainer$position <- 6
+  for(variable in options$variables) {
+    if(!is.null(subcontainer[[variable]]))
+      next
+    descriptivesPlotRainCloud <- createJaspPlot(title = variable, width = 480, height = 320)
+    descriptivesPlotRainCloud$dependOn(optionContainsValue = list(variables = variable))
+    subcontainer[[variable]] <- descriptivesPlotRainCloud
+    if(ready){
+      p <- try(.ttestIndependentDescriptivesRainCloudPlotFill(dataset, options, variable))
+      if(isTryError(p))
+        descriptivesPlotRainCloud$setError(.extractErrorMessage(p))
+      else
+        descriptivesPlotRainCloud$plotObject <- p
+    }
+  }
+  return()
+}
+
+.ttestIndependentDescriptivesRainCloudPlotFill <- function(dataset = NULL, options, variable) {
```

Comment:
why do you have the default `dataset = NULL`? Not necessarily a mistake, just asking for the reason

Reply:
Because having to default let to problems in a PR for previous plots in the regression module. But other than that there is not  a specific reason for it. I can remove it, if that is better?